source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -361,7 +361,7 @@ absl::Status ProcessorState::handleTrailersResponse(const TrailersResponse& resp
     if (response.has_header_mutation()) {
       auto mut_status = MutationUtils::applyHeaderMutations(
           response.header_mutation(), *trailers_, false, filter_.config().mutationChecker(),
-          filter_.stats().rejected_header_mutations_);
+          filter_.stats().rejected_header_mutations_, body_mode_ == ProcessingMode::STREAMED);
```

Comment:
same as above, change the last parameter to " body_mode_!= ProcessingMode::NONE"